### PR TITLE
fix(server): merge existing and new configured properties on server load

### PR DESCRIPTION
**Pull request by squakez**
* Added a merge method to include existing and new configured properties instead of replacing the one coming from previous connector's version
* Added a couple of unit test to validate the merge

Ref #8466

### CODE REVIEW

File: app/server/dao/src/main/java/io/syndesis/server/dao/manager/DataManager.java [hunk 1]
```diff
@@ -167,9 +168,9 @@ private void loadData() {
                         final Connector existing = fetch(Connector.class, id);
                         if (existing != null) {
                             // the only mutable part of the Connector
-                            final Map<String, String> configuredProperties = existing.getConfiguredProperties();
-
-                            connector = connector.builder().configuredProperties(configuredProperties).build();
+                            final Map<String, String> existingConfiguredProperties = existing.getConfiguredProperties();
+                            final Map<String, String> mergedConfiguredProperties = merge(existingConfiguredProperties, connector.getConfiguredProperties());
+                            connector = connector.builder().configuredProperties(mergedConfiguredProperties).build();
```
Comment (zregvart) [inline app/server/dao/src/main/java/io/syndesis/server/dao/manager/DataManager.java:173]: I've upgraded Immutables for #8269 and there's a `putAllConfiguredProperties` method on `Builder` classes, not sure if it's there in the current version. Might want to wait for that as it makes the implementation simpler:

```java
connector = connector.builder().putAllConfiguredProperties(existing.getConfiguredProperties())
```

File: app/server/dao/src/main/java/io/syndesis/server/dao/manager/DataManager.java [hunk 2]
```diff
@@ -182,6 +183,20 @@ private void loadData() {
         }
     }
 
+    private static Map<String, String> merge(Map<String, String> existingProps, Map<String, String> newProps) {
+        Map<String, String> mergedProps = new HashMap<>();
+        mergedProps.putAll(existingProps);
+        // New properties will override any existing property with same name
+        mergedProps.putAll(newProps);
+        LOGGER.info("Merging {} new configured properties with {} existing configured properties: {} merged configured properties",
```
Comment (zregvart) [inline app/server/dao/src/main/java/io/syndesis/server/dao/manager/DataManager.java:191]: Not sure if this merits an INFO level log

File: app/server/dao/src/main/java/io/syndesis/server/dao/manager/DataManager.java [hunk 3]
```diff
@@ -182,6 +183,20 @@ private void loadData() {
         }
     }
 
+    private static Map<String, String> merge(Map<String, String> existingProps, Map<String, String> newProps) {
```
Comment (zregvart) [inline app/server/dao/src/main/java/io/syndesis/server/dao/manager/DataManager.java:186]: There's a `io.syndesis.common.util.CollectionsUtils::aggregate` you can use instead.

File: app/server/dao/src/main/java/io/syndesis/server/dao/manager/DataManager.java (hunk 1 above)
Comment (squakez) [reply app/server/dao/src/main/java/io/syndesis/server/dao/manager/DataManager.java:173]: Unfortunately it fails the `shouldOverrideExistingConfiguredProperties` test. Basically it does not override any new configured properties introduced in the new connector version.

File: app/server/dao/src/main/java/io/syndesis/server/dao/manager/DataManager.java (hunk 2 above)
Comment (squakez) [reply app/server/dao/src/main/java/io/syndesis/server/dao/manager/DataManager.java:191]: I could lower to `DEBUG`. However I feel like it is something we should notify to a user: in my case, if the info was present, I had not to go deep in code to understand that there was a merge of properties that was the root cause of the failure.

File: app/server/dao/src/main/java/io/syndesis/server/dao/manager/DataManager.java (hunk 3 above)
Comment (squakez) [reply app/server/dao/src/main/java/io/syndesis/server/dao/manager/DataManager.java:186]: Many tests failing ans it does not allow properties duplication: `java.lang.IllegalStateException: Duplicate key smtp` when calling the `loadData()` method.

**Pull request merged**